packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -104,12 +104,15 @@ async function getTransitiveDependencies(packagePath, isExternal) {
             .map(dependency => dependency[0]);
 
         for (const dependency of dependencies) {
+            if (isExternal(dependency)) {
```

Comment:
Are the changes in this file responsible for the following things you mentioned in OP? If not, can you point me to them?
```
- deleted stale dependency react-native-device-info from NMR (is external)
- deleted lib/commonsjs + lib/module build variants of react-native-image-picker from NMR. These were stale artefacts. Metro uses the .ts.
```

Reply:
I can see how this was not clear, apologies. 

No, these changes were caused by:
- in automation, delete js actions from the Native Components Test Project before copying over everything (once compiled) from widget-resources. This eliminates any code/dependencies that have become stale ("become stale" = widget-resources *used* to use this code/dependency, but no longer does)

Reply:
Previously, the automation would only *add* code/dependencies. Meaning we got into a situation where code/deps were commited to the Native Components Test Project but not used.

Reply:
> delete js actions from the Native Components Test Project before copying over everything (once compiled) from widget-resources

Got it, thanks!